news/12079.bugfix.rst

```diff
@@ -0,0 +1 @@
+Fix slowness when using ``importlib.metadata`` and there is a large overlap between already installed and to-be-installed packages.
```

Comment:
Should probably still mention 3.11, where `imporlib.metadata` is the default and impacts most people

Reply:
Done, and I also mention the `_PIP_USE_IMPORTLIB_METADATA` env variable that can override this.
I am happy for you to edit the news snippet directly, if that helps.